modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -285,20 +286,22 @@ template<typename Impl, typename... Ins>
 struct OCVSetupHelper<Impl, std::tuple<Ins...>>
 {
     template<int... IIs>
-    static void setup_impl(const GMetaArgs &metaArgs, const GArgs &args, GArg &state,
+    static void setup_impl(const GMetaArgs &metaArgs, const GArgs &args,
+                           const GCompileArgs &compileArgs, GArg &state,
                            detail::Seq<IIs...>)
     {
         // TODO: unique_ptr <-> shared_ptr conversion ?
         // To check: Conversion is possible only if the state which should be passed to
         // 'setup' user callback isn't required to have previous value
         std::shared_ptr<typename Impl::State> stPtr;
-        Impl::setup(detail::get_in_meta<Ins>(metaArgs, args, IIs)..., stPtr);
+        Impl::setup(detail::get_in_meta<Ins>(metaArgs, args, IIs)..., compileArgs, stPtr);
```

Comment:
I'd put compile arguments after the state pointer.

ALSO:

I'd allow to let people write both versions of setup -- with compile args or not.
The extended version (with compile args) is the callback triggered by runtime.
The reduced (current master) version can be called from that default implementation.

Properly implementing this will require SFINAE checks which you can implement as a separate PR.
So far we could go with a less correct implementation where one callback just calls another by default.

Reply:
Fixed, please check if such implementation you can accept current implementation